java-squid/src/main/java/org/sonar/java/model/expression/MethodInvocationTreeImpl.java

```diff
@@ -114,10 +115,11 @@ public void accept(TreeVisitor visitor) {
   @Override
   public Iterator<Tree> childrenIterator() {
     return Iterators.concat(
-      Iterators.singletonIterator(methodSelect),
-      Iterators.singletonIterator(typeArguments),
-      arguments.iterator()
-      );
+      typeArguments != null ? Iterators.<Tree>singletonIterator(typeArguments) : Iterators.<Tree>emptyIterator(),
+      Iterators.<Tree>forArray(methodSelect, openParenToken),
+      // FIXME SONARJAVA-547 separators are lost
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)